### PR TITLE
[safebreach-129] SafeBreach BIOC

### PR DESCRIPTION
**Pull request by gtsafeB**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [x] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Packs/SafeBreach/Integrations/SafeBreach_v2/README.md [hunk 1]
```diff
@@ -57,38 +68,25 @@ Every customer environment might have some of the insights depending on the simu
 | 15 |  Email | Hash  | Email with encrypted malicious attachments
 | 24 |  Email |Hash  | Email with non-encrypted malicious attachment 
 
-
-
-
-## Playbooks
```
Comment (reutshal) [inline Packs/SafeBreach/Integrations/SafeBreach_v2/README.md:63]: Why did you remove this section? I see there are some playbooks in the pack.

File: Packs/SafeBreach/Integrations/SafeBreach_v2/SafeBreach_v2.yml [hunk 2]
```diff
@@ -260,15 +265,19 @@ script:
...
+      type: String
   - arguments:
     - default: false
-      description: The ID of the insight to rerun.
-      isArray: false
-      name: insightId
-      required: true
+      description: Array of insight IDs to rerun.
+      isArray: true
+      name: insightIds
```
[the remark this answers is not in the document]
Comment (gtsafeB) [reply Packs/SafeBreach/Integrations/SafeBreach_v2/SafeBreach_v2.yml:275]: Currently, we don't have customers using this integration. we would prefer to change it now.

[the remark this answers is not in the document]
Comment (ronykoz) [reply Packs/SafeBreach/Integrations/SafeBreach_v2/SafeBreach_v2.yml:275]: @gtsafeB how certain are you?
I do think that it would be best to support an array + not changing the name.

File: Packs/SafeBreach/Playbooks/SafeBreach_Process_Non-Behavioral_Insights_Feed.yml [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+      period:
+        by: ""
+        byto: ""
+        byfrom: ""
+        tovalue:
+        fromvalue:
+        field: ""
+      fromdatelicenseval: 0001-01-01T00:00:00Z
+    runFromLastJobTime: true
+outputs: []
```
Comment (David-BMS) [inline Packs/SafeBreach/Playbooks/SafeBreach_Process_Non-Behavioral_Insights_Feed.yml:456]: Shouldn't the playbook have outputs or everything is done just in the sublaybooks?

File: Packs/SafeBreach/Playbooks/SafeBreach_Process_Non-Behavioral_Insights_Feed_README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+| --- | --- | --- | --- |
+| Indicator Query | Indicators matching the indicator query will be used as playbook input | sourceBrands:["SafeBreach*"] and -safebreachisbehavioral:T | Optional |
+
+## Playbook Outputs
+---
+There are no outputs for this playbook.
+
+## Playbook Image
+---
+![SafeBreach - Process Non-Behavioral Insights Feed](Insert the link to your image here)
```
Comment (David-BMS) [inline Packs/SafeBreach/Playbooks/SafeBreach_Process_Non-Behavioral_Insights_Feed_README.md:35]: Missing link for the image.

File: Packs/SafeBreach/Playbooks/SafeBreach_Rerun_Insights_README.md [hunk 5]
```diff
@@ -44,4 +44,6 @@ This playbook uses the following sub-playbooks, integrations, and scripts.
 | SafeBreach.Insight.NetworkDirection | Communication direction of Insight, relative to the target \(inbound/outbound\) | unknown |
 | SafeBreach.Insight.AttacksCount | List of all insight related SafeBreach attack ids | unknown |
 
-<!-- Playbook PNG image comes here -->
\ No newline at end of file
+## Playbook Image
+---
+![SafeBreach - Rerun Insights](Insert the link to your image here)
```
Comment (David-BMS) [inline Packs/SafeBreach/Playbooks/SafeBreach_Rerun_Insights_README.md:49]: Missing image link.

File: Packs/SafeBreach/Playbooks/SafeBreach_Rerun_Single_Insight_README.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+| SafeBreach.Insight.Rerun.AttacksCount | Count of the attacks executed in the insight rerun test. | unknown |
+| SafeBreach.Test.Id | ID of the test. | unknown |
+| SafeBreach.Test.Name | Name of the test. | unknown |
+| SafeBreach.Test.AttacksCount | The number of attacks executed in the insight rerun test. | unknown |
+| SafeBreach.Test.Status | Test run status. For insight rerun, starts from PENDING. | unknown |
+| SafeBreach.Test.ScheduledTime | Time when the test was triggered. | unknown |
+
+## Playbook Image
+---
+![SafeBreach Rerun Single Insight](Insert the link to your image here)
```
Comment (David-BMS) [inline Packs/SafeBreach/Playbooks/SafeBreach_Rerun_Single_Insight_README.md:42]: Missing image link.

**Comment by reutshal** [comment]
@gtsafeB Regarding the integration's test playbook :
Since you have mocked unit tests, and the data returned from SafeBreach will probably keep changing, I suggest to only run the integration commands in the test playbook (without verifying the context).
Any error to exception will fail the test playbook. The rest of it should be taken care of in the UT.

File: Packs/SafeBreach/Playbooks/SafeBreach_Compare_and_Validate_Insight_Indicators.yml [hunk 7]
```diff
@@ -0,0 +1,760 @@
...
+    skipunavailable: false
+    quietmode: 0
+  "12":
+    id: "12"
+    taskid: 993d9910-6e55-4e9b-8fab-147b9db63678
+    type: regular
+    task:
+      id: 993d9910-6e55-4e9b-8fab-147b9db63678
+      version: -1
+      name: 'Map Input Insight to Context '
```
[the remark this answers is not in the document]
Comment (gtsafeB) [reply Packs/SafeBreach/Playbooks/SafeBreach_Compare_and_Validate_Insight_Indicators.yml:589]: The following task assumed this context exist

File: Packs/SafeBreach/Playbooks/SafeBreach_Compare_and_Validate_Insight_Indicators.yml [hunk 8]
```diff
@@ -0,0 +1,760 @@
...
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "11"
+    scriptarguments:
+      append: {}
+      key:
+        simple: IndicatorsBeforeValidationFiltered
```
[the remark this answers is not in the document]
Comment (gtsafeB) [reply Packs/SafeBreach/Playbooks/SafeBreach_Compare_and_Validate_Insight_Indicators.yml:641]: This name helps to understand the logic of the flow

File: Packs/SafeBreach/Playbooks/SafeBreach_Process_Non-Behavioral_Insights_Feed.yml (hunk 3 above)
Comment (gtsafeB) [reply Packs/SafeBreach/Playbooks/SafeBreach_Process_Non-Behavioral_Insights_Feed.yml:456]: it is the main processing playbook it performs all the tasks within it there is no expectation to use this playbook as sub playbook.

File: Packs/SafeBreach/Playbooks/SafeBreach_Compare_and_Validate_Insight_Indicators.yml [hunk 9]
```diff
@@ -0,0 +1,760 @@
...
+        "x": 150,
+        "y": -510
+      }
+    }
+  }
+inputs:
+- key: IndicatorsBefore
+  value: {}
+  required: true
+  description: Indicators extracted before remediation
```
[the remark this answers is not in the document]
Comment (gtsafeB) [reply Packs/SafeBreach/Playbooks/SafeBreach_Compare_and_Validate_Insight_Indicators.yml]: applied fixes to the description

File: Packs/SafeBreach/Playbooks/SafeBreach_Compare_and_Validate_Insight_Indicators.yml [hunk 10]
```diff
@@ -0,0 +1,760 @@
...
+  playbookInputQuery:
+- key: Insight
+  value: {}
+  required: true
+  description: Insight to verify remediation for
```
[the remark this answers is not in the document]
Comment (gtsafeB) [reply Packs/SafeBreach/Playbooks/SafeBreach_Compare_and_Validate_Insight_Indicators.yml]: applied fixes to the description

File: Packs/SafeBreach/Playbooks/SafeBreach_Compare_and_Validate_Insight_Indicators.yml [hunk 11]
```diff
@@ -0,0 +1,760 @@
...
+  playbookInputQuery:
+outputs:
+- contextPath: RemediatedIndicators
+  description: List of indicators that were remediated
+  type: unknown
```
[the remark this answers is not in the document]
Comment (gtsafeB) [reply Packs/SafeBreach/Playbooks/SafeBreach_Compare_and_Validate_Insight_Indicators.yml]: Done.

File: Packs/SafeBreach/Playbooks/SafeBreach_Compare_and_Validate_Insight_Indicators.yml [hunk 12]
```diff
@@ -0,0 +1,760 @@
...
+- contextPath: NotRemediatedIndicators
+  description: List of indicators that were not remediated
+  type: unknown
```
[the remark this answers is not in the document]
Comment (gtsafeB) [reply Packs/SafeBreach/Playbooks/SafeBreach_Compare_and_Validate_Insight_Indicators.yml]: Done.

**Comment by reutshal** [comment]
@gtsafeB What is the status here? Do you need some help fixing the build?

**Comment by reutshal** [comment]
@gtsafeB @David-BMS What's the status here? Can we merge?
